geode-core/src/jpfTest/java/org/apache/geode/internal/cache/FilterProfileConcurrencyTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Set;
+import java.util.concurrent.Future;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
+import org.apache.geode.distributed.internal.membership.MemberAttributes;
+import org.apache.geode.internal.Version;
+import org.apache.geode.internal.cache.tier.InterestType;
+import org.apache.geode.internal.util.BlobHelper;
+import org.apache.geode.test.concurrency.ConcurrentTestRunner;
+import org.apache.geode.test.concurrency.ParallelExecutor;
+
+@RunWith(ConcurrentTestRunner.class)
+public class FilterProfileConcurrencyTest {
+
+  @Test
+  public void serializationOfFilterProfileWithConcurrentUpdateShouldSucceed(
+      ParallelExecutor executor) throws Exception {
+    warmUp();
```

Comment:
Would it make sense to bake warmup into the test framework. Since these should be unit test and good unit tests have no side effects and run in a very short time then wouldn't it be okay to run the test twice, once for warmup and the second inside the JPF context?

Reply:
My intuition tells me it would be nice to have an escape hatch in case we were testing issues that had to do with setting up a Cache or classloading or logging or something.

Reply:
@galen-pivotal good point. So maybe an annotation to `@warmup` or `@nowarmup` depending on the default?

Reply:
I think you are right - the test shouldn't have to do this warmup business. Let me see if I can work something in (with a configuration option to disable it).

Reply:
After more investigation, most of the issues with classloading were actually due to initializing the logger - it updates a whole bunch of shared state. The next biggest issue was InternalDataSerializer static initialization - that updates a bunch of shared state as well.

I completely override the logger classes to just return my own logger that does nothing. I also set the jpf flags that treat static initializers and final initialization as atomic. I added configuration options so that the user can pass in properties configuring JPF, in case they want to change that behavior.

Long story short, the test no longer needs to warm up, and it runs a bit faster than it did before.